### PR TITLE
fix: Convert to AI Semantic Cache LLM examples to `plugin_example`

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

Converting the LLM examples to use `plugin_example` instead for better consistency.
 

DOCU-4060
### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md [hunk 1]
```diff
@@ -25,40 +25,35 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
-          "port": 6379
-        }
-      }
-    }
-  }'
-```
+{% plugin_example %}
+title: Mistral Example
+plugin: kong-inc/ai_semantic-cache
+name: ai_semantic-cache
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md:31]:
```suggestion
name: ai-semantic-cache
```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md [hunk 2]
```diff
@@ -25,40 +25,34 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
-          "port": 6379
-        }
-      }
-    }
-  }'
-```
+{% plugin_example %}
+title: OpenAI Example
+plugin: kong-inc/ai_semantic-cache
+name: ai_semantic-cache
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md:31]:
```suggestion
name: ai-semantic-cache
```

Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md:31]:
```suggestion
plugin: kong-inc/ai-semantic-cache
name: ai-semantic-cache
```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md [hunk 3]
```diff
@@ -25,40 +25,35 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
-        "threshold": 0.1,
-        "redis": {
-          "host": "redis-stack.redis.svc.cluster.local",
-          "port": 6379
-        }
-      }
-    }
-  }'
-```
+{% plugin_example %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md:28]:
```suggestion

<!--vale off-->
{% plugin_example %}
```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md (hunk 1 above)
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md:31]:
```suggestion
plugin: kong-inc/ai-semantic-cache
name: ai-semantic-cache
```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md [hunk 4]
```diff
@@ -25,40 +25,35 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
+    redis:
+      host: redis-stack.redis.svc.cluster.local
+      port: 6379
+targets:
+  - route
+formats:
+  - curl
+  - kubernetes
+{% endplugin_example %}
+
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md:56]:
```suggestion
<!--vale on-->

```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md [hunk 5]
```diff
@@ -25,40 +25,34 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
-        "threshold": 0.1,
-        "redis": {
-          "host": "redis-stack.redis.svc.cluster.local",
-          "port": 6379
-        }
-      }
-    }
-  }'
-```
+{% plugin_example %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md:28]:
```suggestion

<!--vale off-->
{% plugin_example %}
```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md [hunk 6]
```diff
@@ -25,40 +25,34 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
+    threshold: 0.1
+    redis:
+      host: redis-stack.redis.svc.cluster.local
+      port: 6379
+targets:
+  - route
+formats:
+  - curl
+  - kubernetes
+{% endplugin_example %}
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md:55]:
```suggestion
{% endplugin_example %}
<!--vale on-->

```

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md [hunk 7]
```diff
@@ -25,40 +25,34 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
+    strategy: redis
+    threshold: 0.1
+    redis:
+      host: redis-stack.redis.svc.cluster.local
+      port: 6379
+targets:
+  - route
+formats:
+  - curl
+  - kubernetes
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_openai.md:54]:
```suggestion
  - curl
  - konnect
  - yaml
  - kubernetes
  - terraform
```
Why just two formats? These should work in all.

File: app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md [hunk 8]
```diff
@@ -25,40 +25,35 @@ curl -X POST http://localhost:8001/services/ai-semantic-cache/routes \
...
+    strategy: redis
+    threshold: 0.1
+    redis:
+      host: redis-stack.redis.svc.cluster.local
+      port: 6379
+targets:
+  - route
+formats:
+  - curl
+  - kubernetes
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ai-semantic-cache/how-to/llm-provider-integration-guides/_mistral.md:54]:
```suggestion
formats:
  - curl
  - konnect
  - yaml
  - kubernetes
  - terraform
```

**Review by lena-larionova — CHANGES_REQUESTED** [review]

**Review by lena-larionova — APPROVED** [review]
lgtm!

**Comment by cloudjumpercat** [comment]
@lena-larionova Thanks for the review! I pushed a few extra changes that I noticed as well as an update about how to use `plugin_example` to the contributing docs if you wouldn't mind taking a look at that.

File: app/contributing/kong-plugins.md [hunk 9]
```diff
@@ -92,6 +92,86 @@ If you have any diagrams or screenshots that you want to add to your plugin docu
     > *Figure 1: Diagram showing an OAuth2 authentication flow with Keycloak.*
     ```
 
+## Adding plugin examples
+
+You can use `plugin_example` to easily define and format examples using the plugin. 
+
```
Comment (lena-larionova) [inline app/contributing/kong-plugins.md:98]:
```suggestion

The example accepts all plugin config, and you can configure it to output any combination of targets and formats that you need.

Possible targets:
* `service`
* `route`
* `consumer`
* `consumer_group`
* `global`

Possible formats:
* `curl`
* `konnect`
* `yaml`
* `kubernetes`
* `terrafrom`

```
Let's add this info as well.

**Pull request merged**